src/sync/queue.rs

```diff
@@ -22,10 +23,12 @@ pub struct Queue<T> {
 
 #[derive(Debug)]
 struct Node<T> {
-    data: T,
+    data: ManuallyDrop<T>,
```

Comment:
I still don't fully understand the motivation to use `ManuallyDrop` here..  Could you please leave some comments on it?